Engine/DataStream.cs

```diff
@@ -52,14 +52,13 @@ public DataStream(IDataFeed feed)
         {
             do
             {
-                TimeSlice timeSlice;
-                while (_feed.Bridge.TryDequeue(out timeSlice))
+                foreach (var timeSlice in _feed.Bridge.GetConsumingEnumerable())
                 {
                     AlgorithmTime = timeSlice.Time;
                     yield return timeSlice.Data;
                 }
-            }
-            while (!_feed.LoadingComplete && !_feed.Bridge.IsEmpty);
```

Comment:
Why don't we need loading completed anymore? What if data feed completes too quickly (e.g. daily data)